Add empty dataset check in headers method

GamerClassN7 opened the pull request:
## Summary
- prevent headers() from operating on an empty dataset

## Testing
- `php /tmp/test_headers.php`

GamerClassN7 committed:
Add dataset emptiness check in headers

GamerClassN7 committed:
Delegate getHeader to headers

GamerClassN7 committed:
Inline headers generation

GamerClassN7 committed:
Style headers guard and inline getHeader

GamerClassN7 committed:
Refactor UseDatabase trait to remove duplicate column logic

GamerClassN7 committed:
Throw exception for empty dataset in headers

Xinatorus commented on src/Livewire/DataTableComponent.php:
```diff
@@ -83,11 +83,20 @@ public function dataset(): array
+        $data = $this->dataset();
+
+        if (empty($data)) {
+            throw new \RuntimeException('DataTable dataset cannot be empty.');
```
Tim si nejsem jisty, data muzou byt prece prazdne ne?

GamerClassN7 replied:
ano ale potom nemáš header takže je to critical error

Xinatorus commented on src/Traits/UseDatabase.php:
```diff
@@ -77,7 +61,7 @@ public function datasetFromDB($query): array
         }
 
-        $this->itemsTotal = $query->count();
+        $this->itemsTotal = (clone $query)->count();
```
proc clone?

GamerClassN7 replied:
Má to prý delat co s tím jak LV zpracoovává a zpětně propaguej query

GamerClassN7 committed:
Ensure relation sorting uses join aliases

GamerClassN7 committed:
Fix relation sorting alias reuse

GamerClassN7 committed:
DataTableComponent: optimize array path

- Reorder: filter → transform → sort → paginate
- Fix date filter logic and branches
- Cache column transform method lookups
- Replace Collection sort with usort for speed (kept on revert)
- Add dot-key accessor helper
- Rename updatedHeaderfilter() → updatedHeaderFilter()

Also ensure itemsTotal reflects filtered count and pagination slices post-sort.

GamerClassN7 committed:
Merge branch 'codex/add-empty-dataset-check-to-headers-method' of https://github.com/steelants/Livewire-DataTable into codex/add-empty-dataset-check-to-headers-method

Imendin committed:
fix
